client/src/modules/RelatedProducts/RelatedProducts.jsx

```diff
@@ -10,15 +11,20 @@ class RelatedProducts extends React.Component {
     }
   }
 
+  openCompare = (product) => {this.setState({compare: true, compareProduct: product})};
+  closeCompare = (event) => {this.setState({compare: false})};
+
   render () {
     return (
       <div className='carosel'>
         {this.props.relatedInfo.start !== 0 ? <button className='previous nav' onClick={this.props.caroselClickLeft}>&larr;</button> : null}
         <div className='RelatedProducts list'>
           {this.props.products.map((product) => {
-            return <RelatedCard key={product.id} product={product} />
+            return <RelatedCard key={product.id} product={product} open={this.openCompare}/>
           })}
         </div>
+        {!this.state.compare ? null : <CompareProducts close={this.closeCompare} currentProduct={this.props.relatedInfo.currentProduct} compareProduct={this.state.compareProduct}/>}
+        {this.props.relatedInfo.end <= this.props.relatedInfo.relatedProductsList.length ? <button className='next nav' onClick={this.props.caroselClickRight}>&rarr;</button> : null}
```

Comment:
Nice use of a ternary. You might be able to break these down into two or three lines instead of one long line for each ternary, and still use the ternary.